chain/client/src/view_client.rs

```diff
@@ -801,12 +801,20 @@ impl Handler<NetworkViewClientMessages> for ViewClientActor {
                         };
                         ShardStateSyncResponse { header, part: None }
                     }
-                    Err(_) => {
-                        // This case may appear in case of latency in epoch switching.
-                        // Request sender is ready to sync but we still didn't get the block.
-                        info!(target: "sync", "Can't get sync_hash block {:?} for state request header", sync_hash);
-                        ShardStateSyncResponse { header: None, part: None }
+                    Ok(false) => {
+                        warn!(target: "sync", "sync_hash {:?} didn't pass validation, possible malicious behavior", sync_hash);
+                        return NetworkViewClientResponses::NoResponse;
                     }
+                    Err(e) => match e.kind() {
+                        ErrorKind::BlockMissing(_) => {
```

Comment:
This actually means the sync block is garbage collected, which means that sync hash is invalid. Not sure why you need to consider it separately.

Reply:
I see. We cannot separate invalid hash from latency.